### PR TITLE
toolchain updates with *-hid-sections.cmake

### PR DESCRIPTION
**Pull request by headupinclouds**
For default hidden visibility with function-sections + data-sections flags for more granular library stripping:

```
include("${CMAKE_CURRENT_LIST_DIR}/flags/function-sections.cmake")
include("${CMAKE_CURRENT_LIST_DIR}/flags/data-sections.cmake")
include("${CMAKE_CURRENT_LIST_DIR}/flags/hidden.cmake")
```

Also, .gitignore emacs files!

### CODE REVIEW

File: bin/detail/toolchain_table.py [hunk 1]
```diff
@@ -334,6 +342,8 @@ def verify(self):
       Toolchain('libcxx', 'Unix Makefiles'),
       Toolchain('libcxx-hid', 'Unix Makefiles'),
       Toolchain('libcxx-hid-fpic', 'Unix Makefiles'),
+      Toolchain('libcxx-hid-fpic-sections', 'Unix Makefiles'),
```
Comment (ruslo) [inline bin/detail/toolchain_table.py]: should be `libcxx-fpic-hid-sections`

Comment (headupinclouds) [inline bin/detail/toolchain_table.py]: fixed

File: ios-10-0-arm64-dep-8-0-hid-sections.cmake [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_common.cmake")
+
+include(polly_fatal_error)
+
+# Fix try_compile
+set(MACOSX_BUNDLE_GUI_IDENTIFIER com.example)
+set(CMAKE_MACOSX_BUNDLE YES)
+set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer")
+set(CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM "$ENV{XCODE_ATTRIBUTE_DEVELOPMENT_TEAM}")
```
Comment (ruslo) [inline ios-10-0-arm64-dep-8-0-hid-sections.cmake]: not needed

File: ios-10-0-dep-8-0-hid-sections.cmake [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_common.cmake")
+
+include(polly_fatal_error)
+
+# Fix try_compile
+set(MACOSX_BUNDLE_GUI_IDENTIFIER com.example)
+set(CMAKE_MACOSX_BUNDLE YES)
+set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer")
+set(CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM "$ENV{XCODE_ATTRIBUTE_DEVELOPMENT_TEAM}")
```
Comment (ruslo) [inline ios-10-0-dep-8-0-hid-sections.cmake]: not needed

File: ios-10-1-arm64-dep-8-0-hid-sections.cmake [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_common.cmake")
+
+include(polly_fatal_error)
+
+# Fix try_compile
+set(MACOSX_BUNDLE_GUI_IDENTIFIER com.example)
+set(CMAKE_MACOSX_BUNDLE YES)
+set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer")
+set(CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM "$ENV{XCODE_ATTRIBUTE_DEVELOPMENT_TEAM}")
```
Comment (ruslo) [inline ios-10-1-arm64-dep-8-0-hid-sections.cmake]: not needed

File: ios-10-0-arm64-dep-8-0-hid-sections.cmake (hunk 2 above)
Comment (headupinclouds) [inline ios-10-0-arm64-dep-8-0-hid-sections.cmake]: Done.

File: ios-10-0-dep-8-0-hid-sections.cmake (hunk 3 above)
Comment (headupinclouds) [inline ios-10-0-dep-8-0-hid-sections.cmake]: Done.

File: ios-10-1-arm64-dep-8-0-hid-sections.cmake (hunk 4 above)
Comment (headupinclouds) [inline ios-10-1-arm64-dep-8-0-hid-sections.cmake]: Thanks.

File: ios-10-0-arm64-dep-8-0-hid-sections.cmake [hunk 5]
```diff
@@ -27,12 +27,6 @@ include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_common.cmake")
 
 include(polly_fatal_error)
 
-# Fix try_compile
-set(MACOSX_BUNDLE_GUI_IDENTIFIER com.example)
-set(CMAKE_MACOSX_BUNDLE YES)
-set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer")
```
Comment (ruslo) [inline ios-10-0-arm64-dep-8-0-hid-sections.cmake]: Actually this one needed:
```cmake
set(MACOSX_BUNDLE_GUI_IDENTIFIER com.example)
set(CMAKE_MACOSX_BUNDLE YES)
set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer")
```

Comment (headupinclouds) [inline ios-10-0-arm64-dep-8-0-hid-sections.cmake]: Is this needed for all 3?

Comment (ruslo) [inline ios-10-0-arm64-dep-8-0-hid-sections.cmake]: yep

Comment (headupinclouds) [inline ios-10-0-arm64-dep-8-0-hid-sections.cmake]: Done.

**Comment by ruslo** [comment]
Applied with the [few cleanups](https://github.com/ruslo/polly/commit/6b37847f92f0a4a68996d918a11dca006fa61457) and [fix](https://github.com/ruslo/polly/commit/b9b9897522509422c843d6acad22ee06cb3039d4).

**Comment by headupinclouds** [comment]
Thank you.

**Pull request merged**